src/org/mozilla/javascript/NativeArray.java

```diff
@@ -1656,7 +1660,14 @@ private static Scriptable js_concat(
         scope = getTopLevelScope(scope);
         final Scriptable result = cx.newArray(scope, 0);
 
-        long length = doConcat(cx, scope, result, o, 0);
```

Comment:
Does this need to be changed here? I verified this in my environment and the test passed with `doConcat` as well.